# Medical Examiner Case Archives

Cook County (Chicago) medical examiner records, taken from [here](https://datacatalog.cookcountyil.gov/Public-Safety/Medical-Examiner-Case-Archive/cjeq-bs86) after discovery via [Data is Plural](https://tinyletter.com/data-is-plural).

## Do your importing/setup

In [1]:
import pandas as pd
%matplotlib inline

import re

%matplotlib inline

df= pd.read_csv("Medical_Examiner_Case_Archive.csv", na_values=[0, "Data missing"])

## Read in the data, check its row count and column types

In [2]:
df.shape

(22634, 20)

## Cleaning up your data

First you'll want to convert the `Race` and `Gender` columns from codes into actual text to make analysis easier.

### Gender codes

* `-1` - `Data missing`
* `0` - `Female`
* `1` - `Male`
* `2` - `Unknown`

### Race codes

* `-1` - `Data missing`
* `0` - `American Indian`
* `1` - `Asian`
* `2` - `Black`
* `3` - `Other`
* `4` - `Unknown`
* `5` - `White`

In [3]:
df.head()

Case Number        Date of Incident           Date of Death   Age  Gender  \
0  ME2017-04255  09/10/2017 08:28:00 PM  09/10/2017 08:51:00 PM  51.0    Male   
1  ME2018-02149  05/05/2018 06:30:00 AM  05/05/2018 06:41:00 AM  65.0    Male   
2  ME2018-02154  04/26/2018 08:29:00 AM  05/05/2018 04:45:00 AM  95.0    Male   
3  ME2018-02156  05/05/2018 12:00:00 PM  05/05/2018 12:25:00 PM  64.0  Female   
4  ME2018-02158  05/05/2018 09:30:00 AM  05/05/2018 09:50:00 AM  61.0    Male   

    Race  Latino Manner of Death Primary Cause Primary Cause Line B  \
0  White   False             NaN           NaN                  NaN   
1  White   False             NaN           NaN                  NaN   
2  White   False             NaN           NaN                  NaN   
3  White   False             NaN           NaN                  NaN   
4  Black   False             NaN           NaN                  NaN   

  Primary Cause Line C Secondary Cause Gun Related Opioid Related  \
0                  NaN             NaN         NaN            NaN   
1                  NaN             NaN         NaN            NaN   
2                  NaN             NaN         NaN            NaN   
3                  NaN             NaN         NaN            NaN   
4                  NaN             NaN         NaN            NaN   

                  Incident Address Incident City  longitude   latitude  \
0                              NaN           NaN        NaN        NaN   
1               2110 W. 21ST PL #1       CHICAGO -87.678845  41.853279   
2              20704 S Woodlawn Ct     FRANKFORT        NaN        NaN   
3  4008 W. BERTEAU AVENUE,  APT #1       CHICAGO -87.728177  41.957468   
4         3947 W. LEXINGTON STREET       CHICAGO -87.724739  41.871451   

   OBJECTID                   location  
0     20468                        NaN  
1     77827  (41.8532787, -87.6788449)  
2     77839                        NaN  
3     77823  (41.9574677, -87.7281769)  
4     77822  (41.8714515, -87.7247386)

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

## What percent of the dataset is men, and what percent is women?

It should display as **Male** and **Female**, not as numbers.

In [4]:
df.Gender.value_counts() #percentage

Male       16158
Female      6363
Unknown       12
Name: Gender, dtype: int64

## Getting rid of "Data missing"

`Unknown` means that officially the gender or race is unknown, while `Data missing` means the record is incomplete. That means "Data missing" should have been `NaN`!

Go back to your `read_csv` many cells before and make it so that "Data missing" is automatically set as `NaN`.

- *Tip: Do not use `.replace` for this one!*
- *Tip: Look at the options for `read_csv`, there's something that lets you specify missing values*
- *Tip: It isn't `"Data missing"` - think about how you already replaced*
- *Tip: Be sure you're using an array when you tell it what the 'missing' options are*

### After you've done this, re-run all of the the previous cells and confirm that `"Data missing"` does not exist any more

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

## What is the most common race in the dataset? We want percentages.

We'll come back to this later, I'm just having you check the column for now.

In [5]:
df.Race.value_counts() #percentage

White         12490
Black          9366
Asian           342
Other           269
Unknown          28
Am. Indian       15
Name: Race, dtype: int64


## Do a `.value_counts()` on the `Opioid Related` column

In [6]:

df['Opioid Related'].value_counts()


True    3378
Name: Opioid Related, dtype: int64

## That's weird. Did everyone die from opioids? Try again, but including missing data.

In [7]:
df['Opioid Related'].value_counts(dropna=False)

NaN     19256
True     3378
Name: Opioid Related, dtype: int64

## Cleaning up True/False columns

For some reason in this dataset, the True/False columns are either `True` or `NaN`. `NaN` causes a lot of problems, I'd rather have it be false.

You can use [`fillna`](http://pandas.pydata.org/pandas-docs/version/0.22/generated/pandas.DataFrame.fillna.html) to fill in empty data - it's like `.replace` but for `NaN`.

### Replace all `NaN` values with `False` for the "Gun Related" and "Opioid Related" columns.

In [8]:
df['Opioid Related'] = df['Opioid Related'].fillna(value=False)

### Do another value counts on Opioid Related to make sure it has both True and False values

In [9]:
df['Opioid Related'].value_counts(dropna=False)

False    19256
True      3378
Name: Opioid Related, dtype: int64

# Back to analysis!

## What's the average age people were when they died?

In [10]:
df.Age.mean()

52.09132007233273

## Let's look at how the oldest people died

We're just going to browse. Read through how the **oldest 30 people died.**

In [11]:
df.sort_values('Age', ascending = False).head(30)

Case Number        Date of Incident           Date of Death    Age  \
21142  ME2018-00494  01/11/2018 12:00:00 PM  01/24/2018 03:48:00 PM  106.0   
17602  ME2017-02853  06/19/2017 08:15:00 PM  06/20/2017 10:08:00 PM  106.0   
12106  ME2016-03535  07/20/2016 06:30:00 PM  07/20/2016 06:58:00 PM  105.0   
9376   ME2016-00729  02/10/2016 11:53:00 PM  02/11/2016 12:25:00 AM  103.0   
8653   ME2015-05648  12/28/2015 09:30:00 PM  12/28/2015 09:59:00 PM  103.0   
4942   ME2015-01809  04/26/2015 03:30:00 PM  05/03/2015 03:06:00 PM  103.0   
14697  ME2016-06170  12/23/2016 10:30:00 PM  12/24/2016 02:00:00 AM  101.0   
8656   ME2015-05651  12/30/2015 04:30:00 PM  12/30/2015 04:50:00 PM  101.0   
5621   ME2015-02517  06/17/2015 10:30:00 AM  06/17/2015 10:50:00 AM  101.0   
11558  ME2016-02979  06/12/2016 08:30:00 AM  06/17/2016 07:50:00 AM  101.0   
13498  ME2016-04942  10/14/2016 04:00:00 PM  10/14/2016 04:21:00 PM  101.0   
9904   ME2016-01281  03/13/2016 04:30:00 PM  03/13/2016 04:57:00 PM  101.0   
4239   ME2015-01074  03/12/2015 10:00:00 AM  03/12/2015 04:30:00 PM  100.0   
5817   ME2015-02723  06/29/2015 11:00:00 AM  06/29/2015 11:46:00 AM  100.0   
7898   ME2015-04864  10/15/2015 07:41:00 PM  11/10/2015 10:30:00 PM  100.0   
4015   ME2015-00842  02/16/2015 08:34:00 PM  02/26/2015 06:45:00 PM  100.0   
2610   ME2014-01556  11/19/2014 10:41:00 AM  11/20/2014 08:35:00 PM  100.0   
8780   ME2016-00111  12/28/2015 11:11:00 PM  01/03/2016 11:40:00 AM  100.0   
798    ME2018-01708  04/08/2018 03:03:00 PM  04/08/2018 03:12:00 PM  100.0   
3985   ME2015-00812  02/19/2015 12:00:00 PM  02/25/2015 06:44:00 AM  100.0   
17789  ME2017-03051  07/02/2017 03:45:00 PM  07/02/2017 05:40:00 PM  100.0   
17142  ME2017-02376  05/24/2017 04:30:00 PM  05/24/2017 05:47:00 PM  100.0   
9859   ME2016-01233  02/28/2016 10:25:00 AM  03/10/2016 03:20:00 AM  100.0   
13696  ME2016-05143  09/30/2016 12:00:00 PM  10/28/2016 04:08:00 AM  100.0   
10335  ME2016-01724  04/08/2016 06:00:00 AM  04/08/2016 06:47:00 AM  100.0   
14044  ME2016-05503  11/16/2016 09:55:00 AM  11/16/2016 10:00:00 AM  100.0   
17007  ME2017-02248  05/17/2017 12:22:00 PM  05/17/2017 12:50:00 PM   99.0   
914    ME2014-01729  12/03/2014 09:30:00 AM  12/03/2014 10:16:00 AM   99.0   
3674   ME2015-00491  01/28/2015 05:00:00 PM  01/28/2015 05:27:00 PM   99.0   
12048  ME2016-03477  07/16/2016 06:24:00 PM  07/16/2016 06:42:00 PM   99.0   

       Gender   Race  Latino Manner of Death  \
21142  Female  White   False        ACCIDENT   
17602  Female  White   False             NaN   
12106  Female  White   False         NATURAL   
9376   Female  White   False         NATURAL   
8653   Female  Black   False         NATURAL   
4942   Female  White   False        ACCIDENT   
14697  Female  Black   False        ACCIDENT   
8656     Male  White   False         NATURAL   
5621   Female  White   False         NATURAL   
11558  Female  White   False        ACCIDENT   
13498  Female  White   False         NATURAL   
9904     Male  White   False         NATURAL   
4239   Female  White   False        ACCIDENT   
5817     Male  White   False         NATURAL   
7898   Female  White   False        ACCIDENT   
4015   Female  White   False        ACCIDENT   
2610   Female  White   False        ACCIDENT   
8780   Female  White   False        ACCIDENT   
798    Female  White   False         NATURAL   
3985   Female  White   False        ACCIDENT   
17789  Female  White   False        ACCIDENT   
17142  Female  White   False         NATURAL   
9859   Female  Asian   False        ACCIDENT   
13696  Female  Black   False        ACCIDENT   
10335    Male  White   False         NATURAL   
14044    Male  Black   False         NATURAL   
17007  Female  Black   False         NATURAL   
914    Female  Black   False        ACCIDENT   
3674   Female  White   False        ACCIDENT   
12048  Female  Black   False         NATURAL   

                                           Primary Cause Primary Cause Line B  \
21142              

## Seems like a lot of problems with fractures

### What's the median age of someone dying from a cause that involves a fracture?

Are fractures especially dangerous for the elderly?

- *Tip: Filter for a cause that involves a fracture, then take the median age*
- *Tip: If you get a "cannot index NA values" error, the problem is it's trying to search `NaN` values and doesn't know what to do with them. You need to tell pandas to count `NaN` as false by setting another option - it isn't `NaN=False`, but it's close!*

In [12]:
fracture = df[df['Primary Cause'].str.contains('FRACTURE').fillna(value=False)]
fracture

fracture.median()

C:\Users\Shaleen\AppData\Local\Programs\Python\Python37-32\lib\site-packages\numpy\lib\nanfunctions.py:1076: RuntimeWarning: Mean of empty slice
  return np.nanmean(a, axis, out=out, keepdims=keepdims)


Age                  86.000000
Latino                0.000000
Opioid Related        0.000000
longitude           -87.800225
latitude             41.924898
OBJECTID          60827.500000
dtype: float64

### To get a "compared to what?", what's the median age of _anyone_ dying an accidental death?

In [13]:
accident = df[df['Manner of Death'].str.contains('ACCIDENT').fillna(value=False)]
accident
accident.mean()


Age                  52.470464
Latino                0.091648
Opioid Related        0.427429
longitude           -87.734993
latitude             41.850005
OBJECTID          66110.189721
dtype: float64

### What's the median age of each manner of death?

It looks like different kinds of death might happen to different ages of people. Let's investigate that further.

In [14]:
df['Manner of Death'].value_counts()


NATURAL         9220
ACCIDENT        7627
HOMICIDE        2839
SUICIDE         1711
UNDETERMINED     502
PENDING          459
Name: Manner of Death, dtype: int64

In [15]:
natural = df[df['Manner of Death'].str.contains('NATURAL').fillna(value=False)]
natural.median()

Age                  61.000000
Latino                0.000000
Opioid Related        0.000000
longitude           -87.695988
latitude             41.858385
OBJECTID          67794.500000
dtype: float64

In [16]:
accident = df[df['Manner of Death'].str.contains('ACCIDENT').fillna(value=False)]
accident.median()

Age                  51.000000
Latino                0.000000
Opioid Related        0.000000
longitude           -87.715975
latitude             41.871021
OBJECTID          65552.000000
dtype: float64

In [17]:
homicide = df[df['Manner of Death'].str.contains('HOMICIDE').fillna(value=False)]
homicide.median()

Age                  26.000000
Latino                0.000000
Opioid Related        0.000000
longitude           -87.671368
latitude             41.801264
OBJECTID          68575.000000
dtype: float64

In [18]:
suicide = df[df['Manner of Death'].str.contains('SUICIDE').fillna(value=False)]
suicide.median()

Age                  45.000000
Latino                0.000000
Opioid Related        0.000000
longitude           -87.728066
latitude             41.892251
OBJECTID          64436.000000
dtype: float64

### Who is the oldest homicide victim?

It looks like homicide is for young people, so maybe we'll find an interesting outlier?

In [19]:
homicide.sort_values('Age', ascending = False).head(1)

Case Number        Date of Incident           Date of Death   Age  \
3702  ME2015-00519  02/03/2015 08:21:00 AM  02/04/2015 08:34:00 AM  98.0   

      Gender   Race  Latino Manner of Death  \
3702  Female  Black   False        HOMICIDE   

                          Primary Cause Primary Cause Line B  \
3702  MORPHINE AND HYDROCODONE TOXICITY                  NaN   

     Primary Cause Line C                                    Secondary Cause  \
3702                  NaN  HYPERTENSIVE AND ATHEROSCLEROTIC CARDIOVASCULA...   

     Gun Related  Opioid Related Incident Address  Incident City  longitude  \
3702         NaN            True  16300 louis Ave  SOUTH HOLLAND -87.609278   

       latitude  OBJECTID                   location  
3702  41.598109     64404  (41.5981086, -87.6092784)

## Investigating toxicity-related homicides

She was old, and was purposefully overdosed on morphine and hydrocodone. Might have been euthenasia? Let's find similar cases.

### Find every homicide where the primary cause of death is some sort of toxicity

Toxicity can just overdose. You should have **ten rows**.

- *Tip: If you're doing this as one statement, make sure you use your parentheses correctly. If you leave them out, you'll have zero rows*
- *Tip: You could make a homicides-only dataframe if you wanted to*

In [20]:
df[df['Manner of Death'].str.contains('HOMICIDE').fillna(value=False) & df['Primary Cause'].str.contains('TOXICITY').fillna(value=False)]

Case Number        Date of Incident           Date of Death   Age  \
3702   ME2015-00519  02/03/2015 08:21:00 AM  02/04/2015 08:34:00 AM  98.0   
5395   ME2015-02286  05/27/2015 03:15:00 PM  06/03/2015 07:25:00 PM  28.0   
6144   ME2015-03058  07/18/2015 07:00:00 AM  07/19/2015 02:05:00 AM  10.0   
6764   ME2015-03692  08/29/2015 09:41:00 AM  08/29/2015 01:00:00 PM  28.0   
6765   ME2015-03693  08/29/2015 09:41:00 AM  08/29/2015 01:00:00 PM  63.0   
7612   ME2015-04564  10/22/2015 10:39:00 AM  10/22/2015 11:05:00 AM  31.0   
12656  ME2016-04092  08/23/2016 01:37:00 AM  08/23/2016 06:45:00 AM   4.0   
12657  ME2016-04093  08/23/2016 01:37:00 AM  08/23/2016 06:45:00 AM   7.0   
12660  ME2016-04096  08/23/2016 01:37:00 AM  08/23/2016 06:45:00 AM  56.0   
13050  ME2016-04490  09/16/2016 02:45:00 AM  09/16/2016 03:34:00 AM  59.0   

       Gender   Race  Latino Manner of Death  \
3702   Female  Black   False        HOMICIDE   
5395   Female  White   False        HOMICIDE   
6144     Male  Black   False        HOMICIDE   
6764     Male  White    True        HOMICIDE   
6765     Male  Other   False        HOMICIDE   
7612     Male  Other   False        HOMICIDE   
12656  Female  Black   False        HOMICIDE   
12657  Female  Black   False        HOMICIDE   
12660    Male  Black   False        HOMICIDE   
13050    Male  White    True        HOMICIDE   

                                           Primary Cause  \
3702                   MORPHINE AND HYDROCODONE TOXICITY   
5395   COMPLICATIONS OF AMITRIPTYLINE AND METABOLITE,...   
6144                            CARBON MONOXIDE TOXICITY   
6764                            CARBON MONOXIDE TOXICITY   
6765                            CARBON MONOXIDE TOXICITY   
7612                         HEROIN AND ETHANOL TOXICITY   
12656      THERMAL INJURIES AND CARBON MONOXIDE TOXICITY   
12657      THERMAL INJURIES AND CARBON MONOXIDE TOXICITY   
12660      THERMAL INJURIES AND CARBON MONOXIDE TOXICITY   
13050                           CARBON MONOXIDE TOXICITY   

        Primary Cause Line B Primary Cause Line C  \
3702                     NaN                  NaN   
5395                     NaN                  NaN   
6144   INCENDIARY HOUSE FIRE                  NaN   
6764              HOUSE FIRE                  NaN   
6765              HOUSE FIRE                  NaN   
7612                     NaN                  NaN   
12656                    NaN                  NaN   
12657                    NaN                  NaN   
12660                    NaN                  NaN   
13050            GARAGE FIRE                  NaN   

                                         Secondary Cause Gun Related  \
3702   HYPERTENSIVE AND ATHEROSCLEROTIC CARDIOVASCULA...         NaN   
5395                                                 NaN         NaN   
6144                                                 NaN         NaN   
6764                                                 NaN         NaN   
6765   ARTERIOSCLEROTIC CARDIOVASCULAR DISEASE, ACUTE...         NaN   
7612                                                 NaN         NaN   
12656                                                NaN         NaN   
12657                                                NaN         NaN   
12660                                                NaN         NaN   
13050                                                NaN         NaN   

       Opioid Related                Incident Address  Incident City  \
3702             True                 16300 louis Ave  SOUTH HOLLAND   
5395            False  1102 S. Springinsguth  Rd. #1D     SCHAUMBURG   
6144            False                  8327 S. DREXEL        CHICAGO   
6764            False             5209 S Lorel Avenue        CHICAGO   
6765            False                   5205 S. Lorel        CHICAGO   
7612             True     5240 NORTH SHERIDAN APT 417        CHICAGO   
12656           False            8112 S. Essex Avenue        CHICAGO   
12657           F

### Okay, nope, we were wrong.

Those were almost **all from fires**. Apparently homicide is not the best place to go looking for toxicity. What's the most popular manner of death for primary causes involving toxicity?

- *Tip: Remember that `['colname']` is the same as `.colname`. You can't do `.col with spaces` so you'll need to do `['col with spaces']` a lot in this dataset
- *Tip: Or I guess if you really wanted to, you could rename your columns to have spaces in them (IF YOU DO THIS DON'T DO IT IN EXCEL BECAUSE IT WILL PROBABLY BREAK YOUR CSV.)*

In [21]:
df[df['Manner of Death'].str.contains('HOMICIDE').fillna(value=False) & df['Primary Cause'].str.contains('TOXICITY').fillna(value=False)]



Case Number        Date of Incident           Date of Death   Age  \
3702   ME2015-00519  02/03/2015 08:21:00 AM  02/04/2015 08:34:00 AM  98.0   
5395   ME2015-02286  05/27/2015 03:15:00 PM  06/03/2015 07:25:00 PM  28.0   
6144   ME2015-03058  07/18/2015 07:00:00 AM  07/19/2015 02:05:00 AM  10.0   
6764   ME2015-03692  08/29/2015 09:41:00 AM  08/29/2015 01:00:00 PM  28.0   
6765   ME2015-03693  08/29/2015 09:41:00 AM  08/29/2015 01:00:00 PM  63.0   
7612   ME2015-04564  10/22/2015 10:39:00 AM  10/22/2015 11:05:00 AM  31.0   
12656  ME2016-04092  08/23/2016 01:37:00 AM  08/23/2016 06:45:00 AM   4.0   
12657  ME2016-04093  08/23/2016 01:37:00 AM  08/23/2016 06:45:00 AM   7.0   
12660  ME2016-04096  08/23/2016 01:37:00 AM  08/23/2016 06:45:00 AM  56.0   
13050  ME2016-04490  09/16/2016 02:45:00 AM  09/16/2016 03:34:00 AM  59.0   

       Gender   Race  Latino Manner of Death  \
3702   Female  Black   False        HOMICIDE   
5395   Female  White   False        HOMICIDE   
6144     Male  Black   False        HOMICIDE   
6764     Male  White    True        HOMICIDE   
6765     Male  Other   False        HOMICIDE   
7612     Male  Other   False        HOMICIDE   
12656  Female  Black   False        HOMICIDE   
12657  Female  Black   False        HOMICIDE   
12660    Male  Black   False        HOMICIDE   
13050    Male  White    True        HOMICIDE   

                                           Primary Cause  \
3702                   MORPHINE AND HYDROCODONE TOXICITY   
5395   COMPLICATIONS OF AMITRIPTYLINE AND METABOLITE,...   
6144                            CARBON MONOXIDE TOXICITY   
6764                            CARBON MONOXIDE TOXICITY   
6765                            CARBON MONOXIDE TOXICITY   
7612                         HEROIN AND ETHANOL TOXICITY   
12656      THERMAL INJURIES AND CARBON MONOXIDE TOXICITY   
12657      THERMAL INJURIES AND CARBON MONOXIDE TOXICITY   
12660      THERMAL INJURIES AND CARBON MONOXIDE TOXICITY   
13050                           CARBON MONOXIDE TOXICITY   

        Primary Cause Line B Primary Cause Line C  \
3702                     NaN                  NaN   
5395                     NaN                  NaN   
6144   INCENDIARY HOUSE FIRE                  NaN   
6764              HOUSE FIRE                  NaN   
6765              HOUSE FIRE                  NaN   
7612                     NaN                  NaN   
12656                    NaN                  NaN   
12657                    NaN                  NaN   
12660                    NaN                  NaN   
13050            GARAGE FIRE                  NaN   

                                         Secondary Cause Gun Related  \
3702   HYPERTENSIVE AND ATHEROSCLEROTIC CARDIOVASCULA...         NaN   
5395                                                 NaN         NaN   
6144                                                 NaN         NaN   
6764                                                 NaN         NaN   
6765   ARTERIOSCLEROTIC CARDIOVASCULAR DISEASE, ACUTE...         NaN   
7612                                                 NaN         NaN   
12656                                                NaN         NaN   
12657                                                NaN         NaN   
12660                                                NaN         NaN   
13050                                                NaN         NaN   

       Opioid Related                Incident Address  Incident City  \
3702             True                 16300 louis Ave  SOUTH HOLLAND   
5395            False  1102 S. Springinsguth  Rd. #1D     SCHAUMBURG   
6144            False                  8327 S. DREXEL        CHICAGO   
6764            False             5209 S Lorel Avenue        CHICAGO   
6765            False                   5205 S. Lorel        CHICAGO   
7612             True     5240 NORTH SHERIDAN APT 417        CHICAGO   
12656           False            8112 S. Essex Avenue        CHICAGO   
12657           F

### Okay, toxicity deaths (overdoses) are mostly accidents. Let's look at the first 30 accidental deaths involving toxicity.

- *Tip: Remember your parentheses!*

In [22]:
df[df['Primary Cause'].str.contains('TOXICITY').fillna(value=False)].head(30)

Case Number        Date of Incident           Date of Death   Age  \
15   ME2018-01127  03/03/2018 02:15:00 AM  03/03/2018 02:24:00 AM  37.0   
45   ME2018-00910  02/19/2018 08:00:00 PM  02/19/2018 08:22:00 PM  56.0   
51   ME2018-00400  01/20/2018 11:15:00 AM  01/20/2018 11:57:00 AM  19.0   
53   ME2018-00643  02/03/2018 10:45:00 AM  02/03/2018 11:11:00 AM  35.0   
54   ME2018-00671  02/05/2018 12:00:00 AM  02/05/2018 12:10:00 AM  27.0   
65   ME2018-01129  03/03/2018 03:45:00 AM  03/03/2018 04:31:00 AM  34.0   
69   ME2018-01356  03/15/2018 01:31:00 PM  03/15/2018 01:44:00 PM  26.0   
90   ME2017-05713  12/07/2017 09:00:00 AM  12/07/2017 09:33:00 AM  43.0   
93   ME2018-01560  03/29/2018 06:50:00 PM  03/29/2018 07:12:00 PM  38.0   
99   ME2018-01570  03/30/2018 03:33:00 PM  03/30/2018 04:20:00 PM  50.0   
102  ME2018-01574  03/30/2018 10:43:00 PM  03/30/2018 11:03:00 PM  40.0   
106  ME2018-01103  03/01/2018 02:02:00 PM  03/01/2018 02:45:00 PM  68.0   
107  ME2018-02197  05/07/2018 09:27:00 AM  05/08/2018 05:23:00 AM  38.0   
108  ME2017-05182  11/04/2017 08:45:00 PM  11/04/2017 09:03:00 PM  50.0   
109  ME2017-05643  12/02/2017 02:50:00 PM  12/02/2017 02:44:00 PM  45.0   
128  ME2018-02201  04/10/2018 01:14:00 AM  04/14/2018 09:20:00 PM  57.0   
161  ME2018-02220  05/02/2018 12:00:00 PM  05/05/2018 03:17:00 PM  35.0   
175  ME2018-00926  02/20/2018 12:00:00 PM  02/20/2018 12:36:00 PM  39.0   
193  ME2017-04357  09/16/2017 03:46:00 AM  09/16/2017 04:01:00 AM  66.0   
228  ME2016-03209  06/30/2016 09:45:00 AM  06/30/2016 10:02:00 AM  57.0   
232  ME2018-01362  03/16/2018 01:53:00 AM  03/16/2018 02:06:00 AM  49.0   
233  ME2018-01822  04/15/2018 09:00:00 AM  04/15/2018 09:21:00 AM  57.0   
243  ME2018-02302  05/10/2018 05:00:00 PM  05/14/2018 02:50:00 PM  61.0   
252  ME2017-00383  01/22/2017 09:30:00 AM  01/22/2017 10:00:00 AM  53.0   
255  ME2017-06107  12/27/2017 06:18:00 AM  12/30/2017 07:28:00 AM  27.0   
353  ME2018-02380  05/20/2018 09:25:00 AM  05/20/2018 10:03:00 AM  40.0   
356  ME2018-00475  01/23/2018 09:42:00 PM  01/23/2018 09:59:00 PM  53.0   
388  ME2018-01138  03/03/2018 09:30:00 AM  03/03/2018 10:13:00 AM  27.0   
396  ME2018-01232  03/08/2018 02:10:00 PM  03/08/2018 02:42:00 PM  36.0   
398  ME2018-01248  03/09/2018 12:00:00 AM  03/09/2018 11:41:00 AM  45.0   

     Gender   Race  Latino Manner of Death  \
15     Male  White   False        ACCIDENT   
45     Male  Black   False        ACCIDENT   
51     Male  White   False        ACCIDENT   
53     Male  White   False        ACCIDENT   
54     Male  White   False        ACCIDENT   
65     Male  Black   False        ACCIDENT   
69     Male  White   False        ACCIDENT   
90     Male  White   False         SUICIDE   
93     Male  Asian   False        ACCIDENT   
99     Male  White   False        ACCIDENT   
102  Female  White   False        ACCIDENT   
106    Male  Black   False        ACCIDENT   
107  Female  White   False        ACCIDENT   
108    Male  Black   False        ACCIDENT   
109    Male  White   False        ACCIDENT   
128  Female  White   False        ACCIDENT   
161  Female  White   False        ACCIDENT   
175    Male  White    True        ACCIDENT   
193    Male  White   False        ACCIDENT   
228    Male  White   False        ACCIDENT   
232    Male  White   False        ACCIDENT   
233    Male  White   False        ACCIDENT   
243  Female  White   False        ACCIDENT   
252  Female  White   False        ACCIDENT   
255    Male  White   False        ACCIDENT   
353    Male  Black   False        ACCIDENT   
356  Female  Black   False        ACCIDENT   
388    Male  Black   False        ACCIDENT   
396    Male  White    True        ACCIDENT   
398  Female  Black   False        ACCIDENT   

                                         Primary Cause Primary Cause Line B  \
15   COMBINED DRUG (COCAINE, HEROIN, FENTANYL) TOXI...                  NaN   
45         COMBINED DRUG (COCAINE AND HEROIN) TOXICITY                  NaN   
51              

## Wow, that's a lot of drug overdoses. What's more popular for overdosing: heroin, fentanyl, cocaine, or ethanol?

You can count something like "COMBINED ETHANOL, NORDIAZEPAM, AND FENTANYL TOXICITY" under both ethanol and fentanyl.

- *Tip: Search for them individually*

In [23]:
df['Primary Cause'].str.contains('HEROIN').fillna(value=False).value_counts(dropna=False)


False    20612
True      2022
Name: Primary Cause, dtype: int64

In [24]:
df['Primary Cause'].str.contains('FENTANYL').value_counts(dropna=False)


False    21017
True      1493
NaN        124
Name: Primary Cause, dtype: int64

In [25]:
df['Primary Cause'].str.contains('COCAINE').value_counts(dropna=False)


False    21446
True      1064
NaN        124
Name: Primary Cause, dtype: int64

In [26]:
df['Primary Cause'].str.contains('ETHANOL').value_counts(dropna=False)


False    21183
True      1327
NaN        124
Name: Primary Cause, dtype: int64

In [27]:
print('heroin is the most popular overdose')

heroin is the most popular overdose


# Cleaning up Primary Cause

Let's stop investigating for a second and maybe clean up this "Primary Cause" column.

## What are the most common Primary Cause of death? Include `NaN` values

- *Tip: There is an option that keeps `NaN` values when counting things in a column.*

In [28]:
df['Primary Cause'].value_counts(dropna=False)

ORGANIC CARDIOVASCULAR DISEASE                                                                             1983
MULTIPLE GUNSHOT WOUNDS                                                                                    1377
HYPERTENSIVE CARDIOVASCULAR DISEASE                                                                        1177
ARTERIOSCLEROTIC CARDIOVASCULAR DISEASE                                                                     974
HYPERTENSIVE ARTERIOSCLEROTIC CARDIOVASCULAR DISEASE                                                        960
MULTIPLE INJURIES                                                                                           941
HYPERTENSIVE AND ARTERIOSCLEROTIC CARDIOVASCULAR DISEASE                                                    460
PENDING                                                                                                     452
GUNSHOT WOUND OF HEAD                                                                                   

## That was horrible looking. I don't want to read through that - how many `NaN` causes of death are there?

- *Tip: You can use `isnull()` to see if it's missing data, but how do you count the results?*

In [29]:
df['Primary Cause'].isnull().value_counts()

False    22510
True       124
Name: Primary Cause, dtype: int64

## Remove all rows where the primary cause of death has not been filled out.

- *Tip: confirm that you have 22510 rows when you're done*

In [30]:
df = df.dropna(subset=['Primary Cause'])

In [31]:
df['Primary Cause'].dropna()

5                                        MULTIPLE INJURIES
6                                    GUNSHOT WOUND TO HEAD
7                                                  PENDING
8                                                  PENDING
9                                                  PENDING
10                                   GUNSHOT WOUND OF HEAD
11                                       MULTIPLE INJURIES
12                                       MULTIPLE INJURIES
13                         COMPLICATIONS OF MORBID OBESITY
14                                 MULTIPLE GUNSHOT WOUNDS
15       COMBINED DRUG (COCAINE, HEROIN, FENTANYL) TOXI...
16                           MULTIPLE SHARP FORCE INJURIES
17                                    CLOSED HEAD INJURIES
18                          ORGANIC CARDIOVASCULAR DISEASE
19                                 MULTIPLE GUNSHOT WOUNDS
20                        MULTIPLE STAB AND INCISED WOUNDS
21                                                 PENDI

In [32]:
df.shape

(22510, 20)

# Cardiovascular disease

Cardiovascular disease (heart disease) is the number one or number two killer in America.

### Filter for only rows where cardiovascular disease was a primary cause

- *Tip: I hope you know how to deal with the `NaN` error message by now!*

In [33]:
cardi_v = df[df['Primary Cause'].str.contains('CARDIOVASCULAR')]

In [34]:
dc = cardi_v

In [35]:
dc

Case Number        Date of Incident           Date of Death   Age  \
18     ME2018-01135  03/03/2018 09:30:00 AM  03/03/2018 10:05:00 AM  55.0   
27     ME2018-02131  05/03/2018 07:15:00 PM  05/03/2018 08:00:00 PM  48.0   
28     ME2018-02134  05/03/2018 06:20:00 PM  05/03/2018 07:01:00 PM  76.0   
34     ME2018-02170  05/06/2018 02:05:00 PM  05/06/2018 02:46:00 PM  64.0   
43     ME2014-01078  10/18/2014 02:30:00 PM  10/18/2014 03:55:00 PM  67.0   
57     ME2018-02186  05/07/2018 12:26:00 PM  05/07/2018 12:53:00 PM  66.0   
64     ME2018-00995  02/23/2018 09:35:00 AM  02/23/2018 09:50:00 AM  70.0   
66     ME2018-01148  03/03/2018 04:30:00 PM  03/03/2018 05:09:00 PM  58.0   
71     ME2018-01411  03/19/2018 12:00:00 AM  03/19/2018 11:04:00 AM  59.0   
72     ME2018-01910  05/01/2018 10:50:00 AM  05/01/2018 10:59:00 AM  65.0   
77     ME2018-02047  05/04/2018 01:03:00 PM  05/04/2018 01:13:00 PM  77.0   
80     ME2014-00002  08/11/2014 09:35:00 AM  08/11/2014 01:00:00 PM  57.0   
81     ME2014-00003  08/11/2014 09:05:00 AM  08/11/2014 09:40:00 AM  74.0   
84     ME2014-00006  08/10/2014 07:15:00 AM  08/10/2014 08:00:00 AM  82.0   
97     ME2018-01567  03/30/2018 10:49:00 AM  03/30/2018 10:54:00 AM  73.0   
105    ME2018-02195  05/07/2018 11:30:00 PM  05/08/2018 11:57:00 PM  44.0   
122    ME2018-02117  05/03/2018 09:00:00 AM  05/03/2018 09:15:00 AM  78.0   
123    ME2018-02202  05/08/2018 01:26:00 PM  05/08/2018 03:00:00 PM  47.0   
124    ME2018-02207  05/08/2018 01:31:00 PM  05/08/2018 01:49:00 PM  71.0   
135    ME2018-02215  05/08/2018 08:28:00 PM  05/08/2018 09:20:00 PM  47.0   
139    ME2018-02252  05/11/2018 10:55:00 AM  05/11/2018 11:05:00 AM  61.0   
151    ME2018-02159  05/05/2018 07:45:00 PM  05/05/2018 08:10:00 PM  77.0   
170    ME2018-02225  05/09/2018 01:30:00 PM  05/09/2018 01:44:00 PM  70.0   
171    ME2018-02210  05/08/2018 12:00:00 PM  05/08/2018 12:36:00 PM  73.0   
178    ME2018-02230  05/09/2018 03:43:00 PM  05/09/2018 04:35:00 PM  60.0   
191    ME2018-02234  05/09/2018 08:42:00 PM  05/09/2018 10:34:00 PM  58.0   
197    ME2018-02263  05/01/2018 10:00:00 AM  05/01/2018 10:13:00 AM  95.0   
201    ME2018-02213  05/08/2018 09:15:00 PM  05/08/2018 09:35:00 PM  63.0   
209    ME2018-02286  05/13/2018 11:00:00 AM  05/13/2018 11:38:00 AM  81.0   
213    ME2018-02285  05/13/2018 11:30:00 AM  05/13/2018 12:22:00 PM  48.0   
...             ...                     ...                     ...   ...   
22543  ME2018-02011  04/16/2018 12:00:00 AM  04/16/2018 03:37:00 PM  55.0   
22546  ME2018-02014  04/27/2018 03:20:00 PM  04/27/2018 03:42:00 PM  54.0   
22552  ME2018-02019  04/27/2018 07:09:00 PM  04/27/2018 07:18:00 PM  69.0   
22555  ME2018-02037  04/27/2018 12:00:00 AM  04/29/2018 12:24:00 PM  94.0   
22556  ME2018-02038  04/29/2018 12:00:00 PM  04/29/2018 12:23:00 PM  80.0   
22558  ME2018-02040  04/29/2018 01:05:00 PM  04/29/2018 01:37:00 PM  72.0   
22559  ME2018-02041  04/29/2018 03:00:00 PM  04/29/2018 03:51:00 PM  82.0   
22564  ME2018-02046  04/29/2018 06:41:00 PM  04/29/2018 06:50:00 PM  80.0   
22568  ME2018-02057  04/29/2018 07:15:00 PM  04/29/2018 07:25:00 PM  62.0   
22569  ME2018-02058  04/22/2018 12:07:00 PM  04/22/2018 12:44:00 PM  65.0   
22575  ME2018-02065  04/30/2018 03:45:00 PM  04/30/2018 04:00:00 PM  69.0   
22576  ME2018-02066  04/30/2018 11:00:00 AM  04/30/2018 11:14:00 AM  75.0   
22579  ME2018-01213  01/07/2018 07:01:00 PM  01/07/2018 07:41:00 PM  66.0   
22588  ME2018-02135  05/03/2018 08:50:00 PM  05/03/2018 09:22:00 PM  64.0   
22593  ME2018-02140  05/04/2018 09:50:00 AM  05/04/2018 10:44:00 AM  56.0   
22598  ME2018-02082  05/01/2018 10:30:00 AM  05/01/2018 11:15:00 AM  84.0   
22599  ME2018-02083  04/04/2018 07:45:00 PM  04/04/2018 08:05:00 PM  63.0   
22600  ME2018-02084  05/01/2018 02:06:00 PM  05/01/2018 02:30:00 PM  71.0   
22601  ME2018-02085  05/01/2018 04:06:00 PM  05/01/2018 04:20:00 PM  94.0   
22604  ME2018-02088  05/01/2018 05:15:00 PM  05/01/2018 06:50:00 PM  50.0   
2260

### What are the different types?

In [36]:
dc['Primary Cause'].str.extract("(.*) CARDIOVASCULAR")[0].value_counts()

ORGANIC                                                       2007
HYPERTENSIVE                                                  1182
ARTERIOSCLEROTIC                                               998
HYPERTENSIVE ARTERIOSCLEROTIC                                  991
HYPERTENSIVE AND ARTERIOSCLEROTIC                              478
HYPERTENSIVE AND ATHEROSCLEROTIC                               298
ATHEROSCLEROTIC                                                143
HYPERTENSIVE-ARTERIOSCLEROTIC                                   90
ORGANIC                                                         77
HYPERTENSIVE ATHEROSCLEROTIC                                    39
ATHEROSCLEROTIC AND HYPERTENSIVE                                16
ARTERIOSLCEROTIC                                                 9
HYPERTENSIVE AND OBESITY-RELATED                                 4
VALVULAR                                                         4
HYPERTENSIVE ARTERIOSCLEROSIS                                 

### Replace all of those with a nice simple 'CARDIOVASCULAR DISEASE'

- *Tip: you can use `.replace` or `.str.replace`, but they each involve different things! I suggest `.replace`, it looks a little cleaner in this situation*
- *Tip: for `.replace`, you need to give it more options than usual*
c- *Tip: for `.str.replace`, it won't automatically save back into the column, you need to do that yourself*

In [37]:
dc['Primary Cause'].replace(*,'CARDIOVASCULAR', regex = True,inplace=True)

SyntaxError: invalid syntax (<ipython-input-37-8fcf0d54b5bd>, line 1)

### Check the top 5 primary causes. Cardiovascular disease should be first with about 28.4%

We could also clean up gunshots, but... let's just move on.

# The Opioid Epidemic

America has a [big problem with fentanyl](https://www.theatlantic.com/health/archive/2018/05/americas-opioid-crisis-is-now-a-fentanyl-crisis/559445/) and other opioids.

## Find all of the rows where fentanyl was part of the primary cause of death

We don't need `na=False` any more because we *dropped the rows without primary causes*.

In [ ]:
df

## Fentanyl and race

In the late 80's and 90's, the [crack cocaine epidemic](https://en.wikipedia.org/wiki/Crack_epidemic) swept through inner cities in the US. It was treated primarily as a crime problem, while many people say fentanyl and heroin overdoses are being treated as a medical problem due to the racial differences - the crack epidemic mainly affected Black communities, while fentanyl seems to be a problem for everyone.

### How does the racial breakdown of fentanyl deaths compare to the racial breakdown of other causes of death? How about compared to causes of accidental death?

### Now compare it to homicides

### Now compare it to suicide

## These differences seems kind of crazy

Let's look at all of these at once: I want a breakdown of the most common manners of death for **men**, based on race.

Percentages, please, not raw numbers.

You can look at women, too, although I think the numbers are more surprising for men.

## Back to drugs: what is the most popular opioid-related primary cause of death that does NOT involve fentanyl?

- *Tip: Pay attention to your column names! There's one that might tell you if something is opioid-related...*
- *Tip: Usually you can use `not` or `!` to means "not", but for pandas and `.isin` or `.str.contains` you need to use `~`*
- *Tip: For "and" in pandas you'll need to use `&`, and make sure all of your clauses have parens around them, e.g. `df[(df.col1 = 'A') & (df.col2 = 'B')]`.*

# How do heroin and fentanyl deaths compare?

## Count the number of deaths involving heroin, the number of deaths involving fentanyl, and the number of deaths involving both.

- *Tip: This will take 3 different statements*
- *Tip: You should get `813` that include both*

## That's weird.

I heard fentanyl really surpassed heroin in the past few years. Let's see how this 

### Pull the year out and store it in a new column called `year`

If you run `df['Date of Incident'].str.extract("(\d\d\d\d)", expand=False)`, it will pull out the year of each incident. **Store this in a new column called `year`.**

(It's regular expression stuff. `\d\d\d\d` means "four numbers in a row", and `()` + `.str.extract` means "pull it out".)

### What is the datatype of the new `year` column?

## Convert this new column to an integer and save it back on top of itself

- *Tip: This uses is your friend `.astype`*
- *Tip: Make sure to save it back on top of itself!*

## Confirm the column is a number

## Plot the number of opioid deaths by year

If you'd like to make it look nicer, do some sorting and get rid of 2018.

- *Tip: Think of it in a few steps. First, filter for opioid deaths. Then get the number of deaths for each year. Then plot it.*
- *Tip: What's up with 2018? Why's it look so weird? Can you get rid of it? Remember to use lots of parens!*
- *Tip: Make sure the earliest year is on the left. You might need to sort by something other than values.*

## Plot the number of fentanyl deaths by year, and the number of heroin deaths by year

- *Tip: You'll want to look up how to use `ylim` - it will let you set each graphic to use the same scale. This should be separate graphics.*
- *Tip: Pay attention to the numbers on your axes. `sort_index()` will be your friend.*
- *Tip: You should probably get rid of 2018*

## How does this compare to gun deaths?

## But hey: numbers can lie pretty easily!

The numbers are just so low in 2014 and much higher in 2017. What's going on there?

Well, maybe **there just isn't as much data from the earlier years**. Plot how many entries there are for each year.

And we don't know the best way to fix that up yet, so instead I'm going to give you a present.

# Is the true lesson here, don't move to Cook County, Illinois?

Cook County is basically Chicago. It's probably just certain areas that are trouble, right? Let's investigate that without even having a clue how mapping works.

## Fun bonus: Making cheating maps

### Make a new dataframe of every death in the actual city of Chicago

### Confirm this new dataframe has 13,627 rows

### Use lat and long in the worst way possible to make a map

Use `longitude` and `latitude` and `plot` to make a rough map of the city. Chicago [looks like this](https://en.wikipedia.org/wiki/File:DuPage_County_Illinois_Incorporated_and_Unincorporated_areas_Chicago_Highlighted.svg)

- *Tip: Use the `latitude` and `longitude` columns*
- *Tip: You don't want a line graph, of course. Or a bar. What kind is the kind with dots on it?*
- *Tip: Use something like like `figsize=(10,5)` to specify the height and width of the map (but, you know, with better numbers that make it look like chicago)*

## Now let's find out where to live

Make a map of every non-homicide death in Chicago, then plot the homicides on top of it.

Use the `ax=df.plot` trick from the beer cans assignment to plot all of the rows representing homicides vs non-homicides. You can use `color='red'` to make one of them red, and `alpha=0.05` to make each mark very transparent to allow them to layer on top of each other.

## Never tell anyone I let you do that.

But you want to see something actually completely legitimately insane?

**Chicago is one of the most segregated cities in America.** If you'd like to see this for yourself, make a map of `Race`. Plot black vs white in a way similar to what we did above.

Yup.